File: Set 8/Best Time to buy and sell stocks 2/BestTime.cpp
#include <iostream>
#include <vector>
using namespace std;

int maxProfit(vector<int>& prices) {
    int max_profit = 0;
    for (int i = 1; i < prices.size(); i++) {
        if (prices[i] > prices[i - 1]) {
            max_profit += prices[i] - prices[i - 1];
        }
    }
    return max_profit;
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        vector<int> prices(n);
        for (int i = 0; i < n; i++) {
            cin >> prices[i];
        }
        cout << maxProfit(prices) << endl;
    }
    return 0;
}

File: Set 8/Best Time to buy and sell stocks 2/BestTime.py
def maxProfit(prices):
    max_profit = 0
    for i in range(1, len(prices)):
        if prices[i] > prices[i - 1]:
            max_profit += prices[i] - prices[i - 1]
    return max_profit

t = int(input())
for _ in range(t):
    n = int(input())
    prices = list(map(int, input().split()))
    print(maxProfit(prices))